splink/blocking.py

```diff
@@ -40,15 +40,15 @@ def blocking_rule_to_obj(br):
 class BlockingRule:
     def __init__(
         self,
-        blocking_rule: BlockingRule | dict | str,
+        blocking_rule_sql: BlockingRule | dict | str,
```

Comment:
I don't think we should allow this to take as an input any of those types - i think this should be string only